addons/lunch/models/lunch_supplier.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import math
+import pytz
+
+from datetime import datetime, time
+
+from odoo import api, fields, models
+from odoo.osv import expression
+from odoo.tools import float_round
+
+from odoo.addons.base.models.res_partner import _tz_get
+
+
+WEEKDAY_TO_NAME = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday']
+
+def float_to_time(hours, tz=None):
+    """ Convert a number of hours into a time object. """
+    if hours == 24.0:
+        return time.max
+    fractional, integral = math.modf(hours)
+    res = time(int(integral), int(float_round(60 * fractional, precision_digits=0)), 0)
+    if tz:
+        res.replace(tzinfo=pytz.timezone(tz))
+    return res
+
+def time_to_float(t):
+    return float_round(t.hour + t.minute/60 + t.second/3600, precision_digits=2)
+
+class LunchSupplier(models.Model):
+    _name = 'lunch.supplier'
+    _description = 'Lunch Supplier'
+
+    partner_id = fields.Many2one('res.partner', string='Linked partner', required=True)
+
+    name = fields.Char('Name', related='partner_id.name')
+    description = fields.Text('Description')
+
+    email = fields.Char(related='partner_id.email')
+    email_formatted = fields.Char(related='partner_id.email_formatted', readonly=True)
+    phone = fields.Char(related='partner_id.phone')
+    street = fields.Char(related='partner_id.street')
+    street2 = fields.Char(related='partner_id.street2')
+    zip = fields.Char(related='partner_id.zip')
```

Comment:
Since we just have a commit renaming all the named `type` fields into `machin_type` (which break couple of things), maybe we can name this field `zip_code` directly to avoid having the same commit applied for 'zip' as it is also a reserved word in python.